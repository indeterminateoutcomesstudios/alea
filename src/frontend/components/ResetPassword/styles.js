// @flow
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'
import InputBase from 'frontend/components/Input'
import ButtonBase from 'frontend/components/Button'

// @flow
export const Wrapper = styled.div`
  width: 300px;
  align-self: center;
  margin: auto;
  background: ${props => props.theme.background};
  border-radius: 5px;
  padding: 2rem;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
`

export const Heading = styled.h1`
  text-align: center;
  margin: 0;
  margin-bottom: 2rem;
  padding: 1rem;
  line-height: 0;
`

export const Input = InputBase.extend`
  padding: 2rem 1rem;
  background: none;
  border: 1px solid ${props => props.theme.borderColor};
  width: 100%;
  margin-bottom: 2rem;
`

export const Button = ButtonBase.extend`
  display: block;
  width: 100%;
`

export const Error = styled.div`
  margin-bottom: 2rem;
  color: ${colors.red};
  font-style: italic;
`

export const Success = styled.div`
  margin-bottom: 2rem;
  color: ${colors.green};
  font-style: italic;
`
