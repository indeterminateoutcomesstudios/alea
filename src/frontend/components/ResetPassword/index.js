// @flow
import * as React from 'react'
import Logo from 'frontend/components/Logo'
import { Wrapper, Heading, Input, Button, Error, Success } from './styles'

type Props = {
  sendPasswordResetEmail: Function
}

type State = {
  error: string | null,
  success: string | null
}

export default class ResetPasswordNew extends React.Component<Props, State> {
  state = {
    error: null,
    success: null
  }

  username = React.createRef()

  submit = async (e: SyntheticEvent<>) => {
    const { sendPasswordResetEmail } = this.props

    e.preventDefault()
    e.stopPropagation()

    this.setState({ error: null, success: null })

    if (!this.username.current) {
      return
    }

    const username = this.username.current.value

    const result = await sendPasswordResetEmail(username)
    if (result.error === true) {
      if (result.code == 'auth/invalid-email') {
        this.setState({ error: 'The email you entered is invalid.' })
      } else if (result.code == 'auth/user-not-found') {
        this.setState({ error: 'There is no account for the given email.' })
      } else {
        this.setState({ error: 'An unexpected error occurred.' })
      }
    } else if (result.success === true) {
      this.setState({
        success: 'Check your email to complete your password reset.'
      })
    }
  }

  render() {
    const { error, success } = this.state
    return (
      <Wrapper>
        <Heading>
          <Logo height="26px" />
        </Heading>
        <form onSubmit={this.submit}>
          <Input
            name="username"
            type="text"
            placeholder="email"
            innerRef={this.username}
          />
          {error && <Error>{error}</Error>}
          {success && <Success>{success}</Success>}
          <Button large onClick={this.submit}>
            Reset my password
          </Button>
        </form>
      </Wrapper>
    )
  }
}
