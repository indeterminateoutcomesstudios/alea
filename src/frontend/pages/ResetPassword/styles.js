// @flow
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'

export const Background = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${colors.blue};
`
