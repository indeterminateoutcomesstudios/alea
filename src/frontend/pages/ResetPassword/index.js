// @flow
import React from 'react'
import sendPasswordResetEmail from 'frontend/firebase/sendPasswordReset'
import ResetPassword from 'frontend/components/ResetPassword'

export default () => (
  <ResetPassword sendPasswordResetEmail={sendPasswordResetEmail} />
)
