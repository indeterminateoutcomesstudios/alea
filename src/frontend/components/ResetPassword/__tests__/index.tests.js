// @flow
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import ResetPassword from '../index'

describe('Password reset component', () => {
  const renderer = new ShallowRenderer()

  it('renders correctly', () => {
    renderer.render(<ResetPassword sendPasswordResetEmail={() => {}} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
