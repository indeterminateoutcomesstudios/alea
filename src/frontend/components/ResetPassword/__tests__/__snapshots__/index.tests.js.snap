// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password reset component renders correctly 1`] = `
<styled.div>
  <styled.h1>
    <Logo
      height="26px"
    />
  </styled.h1>
  <form
    onSubmit={[Function]}
  >
    <styled.input
      innerRef={
        Object {
          "current": null,
        }
      }
      name="username"
      placeholder="email"
      type="text"
    />
    <styled.button
      large={true}
      onClick={[Function]}
    >
      Reset my password
    </styled.button>
  </form>
</styled.div>
`;
